qutebrowser/utils/utils.py

```diff
@@ -864,6 +864,63 @@ def unused(_arg):
     pass
 
 
+def parse_numsettxt_into_list(txtset, nummax):
```

Comment:
It doesn't freeze for the specific stupid that is `1-100000`, but it does freeze for about a second for this similar stupid `1-1000000` that has only one more zero. Using my ultra-low voltage i7 laptop (i7-5600U CPU @ ~2.6 GHz). 

I'll have a crack at it. It will probably also simplify the reverse function `parse_list_into_numsettxt`.